node/src/components/chain_synchronizer/operations.rs

```diff
@@ -948,34 +954,147 @@ where
     }
 }
 
+#[derive(Debug)]
+pub(crate) enum FetchAndStoreBlockError {
+    Fetcher(FetcherError<BlockAndDeploys>),
+    WrongTypeUnderDeployApprovalsRootHash(Box<StoredValue>),
+    CLValue(CLValueError),
+    Engine(engine_state::Error),
+    BytesRepr(bytesrepr::Error),
+    TrustedStateRootMissing,
+    Impossible(String),
+    NoApprovalsForDeploy(DeployHash),
+}
+
+/// Retrieves the deploy approvals root hash from global state
+async fn fetch_deploy_approvals_root_hash<REv>(
+    height: u64,
+    ctx: &ChainSyncContext<'_, REv>,
+) -> Result<Option<Digest>, FetchAndStoreBlockError>
+where
+    REv: From<NetworkInfoRequest>
+        + From<FetcherRequest<BlockAndDeploys>>
+        + From<StorageRequest>
+        + From<ContractRuntimeRequest>,
+{
+    match ctx
+        .effect_builder
+        .query_global_state(QueryRequest::new(
+            *ctx.trusted_block_header().id().inner(),
```

Comment:
Isn't that the block hash? I believe it should be the state root hash?

Reply:
Thank you, yes :+1: